src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -0,0 +1,553 @@
+                }
+            }
+
+            return modifiedRun;
+        }
+
+        private static bool RenameProperty(JObject jObject, string previousName, string newName)
+        {
+            JToken propertyValue = jObject[previousName];
+            
+            if (propertyValue != null)
+            {
+                jObject.Remove(previousName);
+                jObject[newName] = propertyValue;
+            }
+
+            return propertyValue != null;
+        }
+
+        private static bool UpdateFileHashesProperty(JObject file)
+        {
+            bool modifiedRun = false;
+
+            // https://github.com/oasis-tcs/sarif-spec/issues/243
+            //
+            // file.hashes were previously an array of hash objects.
+            // Now file.hashes is a dictionary where each key is the 
+            // hash algorithm and the value is the hash of the file contents.
+            //
+            // Previously:
+            //     "hashes": [ { "algorithm": "sha-256", "value": "b13ce2678a8807ba0765ab94a0ecd394f869bc81" } ]
+            //
+            // Now:
+            // "hashes": { "sha-256": "b13ce2678a8807ba0765ab94a0ecd394f869bc81" }
+
+            JToken hashes = file["hashes"];
+            if (hashes == null || hashes.Type != JTokenType.Array) { return modifiedRun; }
+
+            JObject rewrittenHashes = new JObject();
+
+            foreach (JObject hash in hashes)
+            {
+                string algorithm = (string)hash["algorithm"];
+                string value = (string)hash["value"];
+
+                rewrittenHashes[algorithm] = value;
+            }
+            file.Remove("hashes");
+            file["hashes"] = rewrittenHashes;
+            modifiedRun = true;
+
+            return modifiedRun;
+        }
+
+        internal static bool UpdateRunNotifications(JArray runs)
+        {
+            bool modifiedRun = false;
+
+            foreach (JObject run in runs)
+            {
+                var invocations = (JArray)run["invocations"];
+
+                if (invocations == null) { continue; }
+
+                foreach (JObject invocation in invocations)
+                {
+                    var notifications = (JArray)invocation["configurationNotifications"];
+                    if (notifications != null) { modifiedRun |= UpdateNotifications(notifications); }
+
+                    notifications = (JArray)invocation["toolNotifications"];
+                    if (notifications != null) { modifiedRun |= UpdateNotifications(notifications); }
+                }
+            }
+
+            return modifiedRun;                
+        }
+
+
+        internal static bool UpdateNotifications(JArray notifications)
+        {
+            bool modifiedNotification = false;
+
+            foreach (JObject notification in notifications)
+            {
+                modifiedNotification |= UpdateNotificationObjects(notification);
```

Comment:
>Objects [](start = 58, length = 7)

You're passing only one object, so it would be `UpdateNotificationObject`, but `UpdateNotification` suffices.